### ISSUE TITLE
[Attention] 超级视客营 MMEngine 🚀🚀🚀

### ISSUE THREAD

**Issue by HAOCHENYE** [open-mmlab/mmengine#732]
![image](https://user-images.githubusercontent.com/57566630/202351166-b68a3161-a363-4403-8e50-f699a19466d1.png)
# 活动介绍

大家好，第一期 OpenMMLab 超级视客营实训活动开始啦！超级视客营实训活动提供十七个方向、上百个不同难度的任务供大家选择，不管你是初涉 AI 的新手还是资深炼丹师，都有适合你的任务供你选择。助力大家上手 OpenMMLab 开源算法库并参与项目建设。本期活动联合北京超级云计算中心，提供算力支持，为大家开发保驾护航。

活动参与方式：选择你感兴趣的任务，在 OpenMMLab 官网提交报名表。完成匹配后，即可和导师对接制定任务规划，开始上手开发。根据不同任务要求在对应的地址提交代码结果，出题方初步 review 通过后即可领取下一个任务或者坐等领奖。活动详情戳：[OpenMMLab 官网活动页](https://openmmlab.com/activity/codecamp)。

# 任务描述

MMEngine 作为 OpenMMLab 的新一代训练框架，将不断吸收和整合有吸引力的前沿技术，然而我们自身的力量是有限的，我们希望建立一个活跃的社区，借助社区的力量让 MMEngine 越来越好。在这次活动的帮助下，我们希望与社区积极互动，开发一些非常有趣的功能。我们对其中的一些任务进行了分级，具体如下。

<table>
<thead>
<th>任务</th>
<th>描述</th>
<th>相关技能</th>
<th>难度</th>
<th>积分</th>
<th>认领状态</th>
<body>
<tr>
  <td align="center">
  支持 Segmind 可视化后端
  </td>
  <td align=left>
  基于 <a href="https://github.com/open-mmlab/mmengine/blob/f9158e233e0357798226254a1f57e0ad7d9e2491/mmengine/visualization/vis_backend.py#L59">BaseVisBackend</a> 实现 Sigmindbackend

  **参考实现**: <a href="https://github.com/open-mmlab/mmcv/blob/master/mmcv/runner/hooks/logger/segmind.py">SegmindLoggerHook</a>

  **步骤1**: 实现 SegmindBackend
  
  **步骤2**: 更新单元测试

  **步骤3**: 在 PR 描述中更新 Segmind 可视化效果

  </td>
  <td>
  Python, PyTorch
  </td>
  <td align="center">⭐️⭐️⭐️</td>
  <td align="center">10</td>
  <td align="center"></td>
</tr>
<tr>
  <td align="center">
  支持 MLFlow 可视化后端
  </td>
  <td align=left>
  基于 <a href="https://github.com/open-mmlab/mmengine/blob/f9158e233e0357798226254a1f57e0ad7d9e2491/mmengine/visualization/vis_backend.py#L59">BaseVisBackend</a> 实现 MLFlowBackend

  **参考实现**: <a href="https://github.com/open-mmlab/mmcv/blob/master/mmcv/runner/hooks/logger/mlflow.py">MlflowLoggerHook</a>

  **步骤1**: 实现 MLFlowBackend
  
  **步骤2**: 更新单元测试

  **步骤3**: 在 PR 描述中更新 MLFlowBackend 可视化效果

  </td>
  <td>
  Python, PyTorch
  </td>
  <td align="center">⭐️⭐️⭐️</td>
  <td align="center">10</td>
  <td align="center">进行中</td>
</tr>

<tr>
  <td align="center">
  支持 Dvclive 可视化后端
  </td>
  <td align=left>
  基于 <a href="https://github.com/open-mmlab/mmengine/blob/f9158e233e0357798226254a1f57e0ad7d9e2491/mmengine/visualization/vis_backend.py#L59">BaseVisBackend</a> 实现 DvcliveBackend

  **参考实现**: <a href="https://github.com/open-mmlab/mmcv/blob/master/mmcv/runner/hooks/logger/dvclive.py">DvcliveLoggerHook</a>

  **步骤1**: 实现 DvcliveBackend
  
  **步骤2**: 更新单元测试

  **步骤3**: 在 PR 描述中更新 DvcliveBackend 可视化效果

  </td>
  <td>
  Python, PyTorch
  </td>
  <td align="center">⭐️⭐️⭐️</td>
  <td align="center">10</td>
  <td align="center"></td>
</tr>

<tr>
  <td align="center">
  支持 Neptune 可视化后端
  </td>
  <td align=left>
  基于 <a href="https://github.com/open-mmlab/mmengine/blob/f9158e233e0357798226254a1f57e0ad7d9e2491/mmengine/visualization/vis_backend.py#L59">BaseVisBackend</a> 实现 NeptuneBackend

  **参考实现**: <a href="https://github.com/open-mmlab/mmcv/blob/master/mmcv/runner/hooks/logger/neptune.py">NeptuneBackend</a>

  **步骤1**: 实现 NeptuneBackend
  
  **步骤2**: 更新单元测试

  **步骤3**: 

  </td>
  <td>
  Python, PyTorch
  </td>
  <td align="center">⭐️⭐️⭐️</td>
  <td align="center">10</td>
  <td align="center">进行中</td>
</tr>

<tr>
  <td align="center">
  支持 ClearML 可视化后端
  </td>
  <td align=left>
  基于 <a href="https://github.com/open-mmlab/mmengine/blob/f9158e233e0357798226254a1f57e0ad7d9e2491/mmengine/visualization/vis_backend.py#L59">BaseVisBackend</a> 实现 ClearMLBackend

  **参考实现**: <a href="https://github.com/open-mmlab/mmcv/tree/master/mmcv/runner/hooks/logger">ClearMLLoggerHook</a>

  **步骤1**: 实现 ClearMLBackend
  
  **步骤2**: 更新单元测试

  **步骤3**: 在 PR 描述中更新 ClearMLBackend 可视化效果

  </td>
  <td>
  Python, PyTorch
  </td>
  <td align="center">⭐️⭐️⭐️</td>
  <td align="center">10</td>
  <td align="center"></td>
</tr>

<tr>
  <td align="center">
  支持 ProfilerHook
  </td>
  <td align=left>
  基于 <a href="https://github.com/open-mmlab/mmengine/blob/f9158e233e0357798226254a1f57e0ad7d9e2491/mmengine/hooks/hook.py#L7">Hook</a> 实现 ProfilerHook

  **参考实现**: <a href="https://github.com/open-mmlab/mmcv/blob/master/mmcv/runner/hooks/profiler.py">ProfilerHook</a>

  **步骤1**: 实现 ProfilerHook
  
  **步骤2**: 更新单元测试

  **步骤3**: 在 PR 描述中更新 ProfilerHook 可视化效果

  </td>
  <td>
  Python, PyTorch
  </td>
  <td align="center">⭐️⭐️⭐️</td>
  <td align="center">10</td>
  <td align="center">进行中</td>
</tr>

<tr>
  <td align="center">
  支持提前终止训练
  </td>
  <td align=left>
  支持当评价指标到达阈值后自动停止训练.

  **步骤1**: 设计并实现提前终止训练
  
  **步骤2**: 更新单元测试

  </td>
  <td>
  Python, PyTorch
  </td>
  <td align="center">⭐️⭐️⭐️</td>
  <td align="center">10</td>
  <td align="center"><a href=https://github.com/open-mmlab/mmengine/pull/739>进行中</a></td>
</tr>

<tr>
  <td align="center">
  支持 ApexOptimWrapper 优化器封装
  </td>
  <td align=left>
  基于 <a href="https://github.com/open-mmlab/mmengine/blob/f9158e233e0357798226254a1f57e0ad7d9e2491/mmengine/optim/optimizer/optimizer_wrapper.py#L16">OptimWrapper</a> 实现 ApexOptimWrapper

  **步骤1**: 实现 ApexOptimWrapper
  
  **步骤2**: 更新单元测试

  </td>
  <td>
  Python, PyTorch
  </td>
  <td align="center">⭐️⭐️⭐️⭐️</td>
  <td align="center">30</td>
  <td align="center"><a href=https://github.com/open-mmlab/mmengine/pull/742>进行中</a></td>
</tr>

<tr>
  <td align="center">
  支持 ReduceLROnPlateau 参数调度策略
  </td>
  <td align=left>
  基于 <a href="https://github.com/open-mmlab/mmengine/blob/f9158e233e0357798226254a1f57e0ad7d9e2491/mmengine/optim/scheduler/param_scheduler.py#L25">ParamScheduler</a> 实现 ReduceLROnPlateau

  **参考实现**: <a href="https://pytorch.org/docs/stable/generated/torch.optim.lr_scheduler.ReduceLROnPlateau.html#torch.optim.lr_scheduler.ReduceLROnPlateau">torch.optim.ReduceLROnPlateau</a>

  **步骤1**: 实现 ReduceLROnPlateauParamScheduler、ReduceLROnPlateauLrScheduler、ReduceLROnPlateauMomentumScheduler
  
  **步骤2**: 更新单元测试

  **步骤3**: 在 PR 描述中更新 ReduceLROnPlateau 参数调度策略可视化效果

  </td>
  <td>
  Python, PyTorch
  </td>
  <td align="center">⭐️⭐️⭐️⭐️</td>
  <td align="center">30</td>
  <td align="center">进行中</td>
</tr>

</thead>
</table>

活动报名地址：[报名表地址](https://openmmlab.com/activity/codecamp/apply)
根据任务难度可以获得对应积分，兑换不同奖品。另外完成任务后在知识社区发布学习心得即可获得额外积分（记得主动找小助手领取哦）。
活动交流群：群二维码

![image](https://user-images.githubusercontent.com/57566630/202354208-bf9646f8-0f79-4a7f-b083-f7ff94e0da39.png)

有任何疑问欢迎大家加入群聊或者 Issue 下参与讨论，快来完成挑战，加入 [OpenMMLab 贡献者队伍](https://openmmlab.com/contributor/path)吧～
